internal/oci/store/store.go

```diff
@@ -78,6 +78,12 @@ const (
 	errOpenLayer        = "cannot open layer"
 	errStatLayer        = "cannot stat layer"
 	errCheckExistence   = "cannot determine whether layer exists"
+	errFmtTooManyLayers = "image has too many layers: %d (max %d)"
+)
+
+var (
+	// MaxLayers is the maximum number of layers an image can have.
+	MaxLayers = 256
```

Comment:
I wonder if this is also a concern for the package manager?

Haven't checked the code, but iirc it will look for a layer within an image that is specifically annotated to contain the package contents before falling back to pulling the entire image.

Reply:
Yes it is, but I'll address that in a separate PR